src/Stratis.Bitcoin/P2P/Peer/NetworkPeer.cs

```diff
@@ -734,39 +766,62 @@ public Task SendMessageAsync(Payload payload)
         /// <param name="System.OperationCanceledException">The node has been disconnected, or the cancellation token has been set to canceled.</param>
         public void SendMessage(Payload payload, CancellationToken cancellation = default(CancellationToken))
         {
+            this.logger.LogTrace("({0}:'{1}')", nameof(payload), payload);
+
             try
             {
                 SendMessageAsync(payload).Wait(cancellation);
             }
             catch (AggregateException aex)
             {
-                ExceptionDispatchInfo.Capture(aex.InnerException).Throw();
+                this.logger.LogTrace("Exception occurred: {0}", aex.InnerException.ToString());
```

Comment:
Maybe use LogError here.

Reply:
i would agree if we have separated logging from console output
any LogError goes to the console in our implementation
and i believe the original NBitcoin code had no console outputs, so to be equivalent with that I try not to log over Debug level unless i'm sure such an exception would not be raised under normal situation. but here aggregate exception can be thrown under very normal circumstances in which LogError would pollute console output when nothing went wrong

Reply:
ah I didn't realize that this exception was occurring regularly and doesn't constitute a non-expected exception.